### DISCUSSION TITLE
500 status getting collected shows for some users

### DISCUSSION THREAD

**Discussion by UweTrottmann** [trakt/trakt-api#338] [Q&A] [answered]
Since 2022-03-18, increasing a lot since 2022-03-24 for at least 31 users getting collected shows ([`sync/collection/shows`](https://trakt.docs.apiary.io/#reference/sync/get-collection/get-collection)) from Trakt fails with HTTP status 500 (with no status message).

Affects
- various device models 
- various Android versions
- various SeriesGuide versions

One affected user shared their Trakt username (if needed, I can send it via email or what you prefer, do not want to post it publicly). I could also get a 500 error on the `users/<slug>/collection` endpoint (as `sync` requires access to the account), this is the test request:
```
--> GET https://api.trakt.tv/users/REDACTED/collection/shows h2
trakt-api-key: REDACTED
trakt-api-version: 2
Authorization: Bearer REDACTED
Host: api.trakt.tv
Connection: Keep-Alive
Accept-Encoding: gzip
User-Agent: okhttp/3.14.9
--> END GET
<-- 500 https://api.trakt.tv/users/REDACTED/collection/shows (1138ms)
date: Wed, 30 Mar 2022 15:59:17 GMT
content-type: application/json
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
x-download-options: noopen
x-permitted-cross-domain-policies: none
referrer-policy: strict-origin-when-cross-origin
x-private-user: false
cache-control: no-cache, no-store
x-error-message: Internal server error. Please open a support issue and include your full API request and response, including all headers.
vary: Accept-Encoding
content-encoding: gzip
x-ratelimit: {"name":"AUTHED_API_GET_LIMIT","period":300,"limit":1000,"remaining":999,"until":"2022-03-30T16:00:00Z"}
x-request-id: 488a02fe-9ba7-4489-b211-0591a10db480
x-runtime: 1.005035
cf-cache-status: DYNAMIC
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 6f420d2c9fb49b71-FRA


<-- END HTTP (0-byte, 20-gzipped-byte body)
```

Let me know if I can share any additional info, or if this is something done wrong on the client side.

https://github.com/UweTrottmann/SeriesGuide/issues/849

**Comment by rectifyer** [comment]
Yes, please email me the username so I can reproduce. You can email me directly or our general support@ email too.

**Reply by UweTrottmann** [reply]
Done via uwe AT seriesgui.de at your email, also sent another affected account. (Sorry, forgot to update here, yesterday.)

**Reply by rectifyer** [reply]
Email received. I plan on digging into this on Monday and will figure out the bug.

**Comment by rectifyer** [comment, marked as answer]
I just pushed a fix for this bug, please let me know if it's working ok.

**Reply by UweTrottmann** [reply]
Last report of a 500 error on 2022-04-04, none since then, so looking good. Emailed users that reached out today, will update here if issue persists. Thanks for resolving this!

**Reply by UweTrottmann** [reply]
One user has also confirmed the fix as working.
